hazelcast/src/main/java/com/hazelcast/internal/partition/impl/MigrationManager.java

```diff
@@ -632,19 +632,24 @@ public void run() {
             }
             partitionServiceLock.lock();
             try {
-                triggerRepartitioningWhenClusterStateAllowsMigration = !isMigrationAllowedByClusterState();
-                if (triggerRepartitioningWhenClusterStateAllowsMigration && logger.isFineEnabled()) {
-                    logger.fine("Migrations are not allowed yet, "
-                            + "repartitioning will be triggered when cluster state allows migrations.");
+                if (!node.getClusterService().getClusterState().isMigrationAllowed()) {
+                    triggerRepartitioningWhenClusterStateAllowsMigration = true;
+                    if (logger.isFineEnabled()) {
+                        logger.fine("Migrations are not allowed yet, "
+                                + "repartitioning will be triggered when cluster state allows migrations.");
+                    }
+                    assignCompletelyLostPartitions();
```

Comment:
I like the change - the separation of behaviour when migrations are not allowed is much clearer now. On the other hand, there is a slight behaviour change - previously, the `isRepartitioningAllowed` call inside the `repartition` method would also trigger the control task, which we don't do anymore. Is this ok?